src/Type/User/Mutation/UserMutation.php

```diff
@@ -240,35 +243,66 @@ public static function update_additional_user_object_data( $user_id, $input, $mu
 	 * @param array $roles   List of roles that need to get added to the user
 	 *
 	 * @access private
+	 * @throws /\Exception
 	 */
 	private static function add_user_roles( $user_id, $roles ) {
 
-		if ( empty( $roles ) || ! is_array( $roles ) ) {
+		if ( empty( $roles ) || ! is_array( $roles ) || ! current_user_can( 'edit_user', $user_id ) ) {
 			return;
 		}
 
 		$user = get_user_by( 'ID', $user_id );
 
 		if ( false !== $user ) {
+
 			foreach ( $roles as $role ) {
-				self::verify_user_role( $role );
-				$user->add_role( $role );
+
+				$verified = self::verify_user_role( $role, $user_id );
+
+				if ( true === $verified ) {
+					$user->add_role( $role );
+				} else if ( is_wp_error( $verified ) ) {
+					$message = $verified->get_error_message();
+					throw new \Exception( $message );
+				} else if ( false === $verified ) {
+					throw new \Exception( __( 'This role cannot be added to this user', 'wp-graphql' ) );
```

Comment:
Should probably specify _which_ user role could not be added here for a clearer message